build/verify-tools.ts

```diff
@@ -0,0 +1,93 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+const hasha = require('hasha');
+const mkdirp = require('mkdirp');
+const semver = require('semver');
+const request = require('request');
+const fs = require('fs-extra');
+const path = require("path");
+const readline = require('readline');
+const progress = require('request-progress');
+const platform = require('../out/src/util/platform');
+const configData = require('../requirements.json');
+
+async function verifyTools() {
+  for (const key in configData) {
+    for (const os in configData[key].platform) {
+      const toolLocations = path.resolve(__dirname, '..', 'cache', os, configData[key].platform[os].dlFileName);
+      const targetFolder =  path.resolve(__dirname, '..', 'cache', os);
+      if (fs.existsSync(toolLocations)) {
```

Comment:
No need to check if file exists, always download and verify checksum. 
Throw exception if  checksum for downloaded file is not matching with one from download configuration.